src/Features/Core/Portable/Completion/Providers/TypeImportCompletionService.cs

```diff
@@ -0,0 +1,349 @@
+                    CancellationToken cancellationToken)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    containingNamespace = ConcatNamespace(containingNamespace, symbol.Name);
+
+                    foreach (var memberNamespace in symbol.GetNamespaceMembers())
+                    {
+                        VisitNamespace(memberNamespace, containingNamespace, builder, cancellationToken);
+                    }
+
+                    var overloads = PooledDictionary<string, TypeOverloadInfo>.GetInstance();
+                    var memberTypes = symbol.GetTypeMembers();
+
+                    // Iterate over all top level internal and public types, keep track of "type overloads".
+                    foreach (var memberType in memberTypes)
+                    {
+                        if (IsAccessible(memberType.DeclaredAccessibility) && memberType.CanBeReferencedByName)
+                        {
+                            overloads.TryGetValue(memberType.Name, out var overloadInfo);
+                            overloads[memberType.Name] = overloadInfo.Aggregate(memberType);
+                        }
+                    }
+
+                    foreach (var pair in overloads)
+                    {
+                        var overloadInfo = pair.Value;
+
+                        // Create CompletionItem for non-generic type overload, if exists.
+                        if (overloadInfo.NonGenericOverload != null)
+                        {
+                            var item = TypeImportCompletionItem.Create(overloadInfo.NonGenericOverload, containingNamespace);
+                            var isPublic = overloadInfo.NonGenericOverload.DeclaredAccessibility == Accessibility.Public;
+                            builder.Add(new CompletionItemEntry(item, isPublic));
+                        }
+
+                        // Create one CompletionItem for all generic type overloads, if there's any.
+                        // For simplicity, we always show the type symbol with lowest arity in CompletionDescription
+                        // and without displaying the total number of overloads.
+                        if (overloadInfo.BestGenericOverload != null)
+                        {
+                            // If any of the generic overloads is public, then the completion item is considered public.
+                            var item = TypeImportCompletionItem.Create(overloadInfo.BestGenericOverload, containingNamespace);
+                            var isPublic = overloadInfo.ContainsPublicGenericOverload;
+                            builder.Add(new CompletionItemEntry(item, isPublic));
+                        }
+                    }
+
+                    overloads.Free();
+                }
+
+                // For top level types, default accessibility is `internal`
+                static bool IsAccessible(Accessibility declaredAccessibility) =>
```

Comment:
i'm a little confused why this method is necessary.  top level types are always 'accessible' in that they can only be internal or public, and you're storing the info for them in either of these cases.  it'sonly later that you filter our 'internal' depending on if you think you have IVT access or not.

Reply:
You are right